libs/core/src/lib/overflow-layout/overflow-layout.component.ts

```diff
@@ -0,0 +1,403 @@
+    /** @hidden */
+    private readonly _subscription = new Subscription();
+
+    /** @hidden */
+    private _overflowPopoverContent: OverflowPopoverContent;
+
+    private _fillTrigger$ = new Subject<void>();
+
+    /** @hidden */
+    private _dir: 'rtl' | 'ltr' = 'ltr';
+
+    private _maxVisibleItems = Infinity;
+
+    /** @hidden */
+    constructor(
+        private _cdr: ChangeDetectorRef,
+        private _zone: NgZone,
+        private _elRef: ElementRef,
+        @Optional() private _rtlService: RtlService
+    ) {
+        this._subscription.add(this._fillTrigger$.pipe(debounceTime(30)).subscribe(() => this._fitVisibleItems()));
+    }
+
+    /**
+     * Triggers layout recalculation of the items.
+     */
+    triggerRecalculation(): void {
+        this._fillTrigger$.next();
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._subscription.unsubscribe();
+    }
+
+    /** @hidden */
+    @HostListener('keyup', ['$event'])
+    private _keyUpHandler(event: KeyboardEvent): void {
+        if (KeyUtil.isKeyCode(event, TAB)) {
+            const index = this._allItems.findIndex(
+                (item) => item.overflowItem?.focusable && item.elementRef.nativeElement === event.target
+            );
+            if (index !== -1) {
+                this._keyboardEventsManager.setActiveItem(index);
+            }
+        }
+
+        if (KeyUtil.isKeyCode(event, [DOWN_ARROW, UP_ARROW, LEFT_ARROW, RIGHT_ARROW])) {
+            event.preventDefault();
+
+            // passing the event to key manager so, we get a change fired
+            this._keyboardEventsManager.onKeydown(event);
+        }
+    }
+
+    /**
+     * Sets current focused element.
+     * @param element Element that needs to be focused.
+     */
+    setFocusedElement(element: OverflowItem): void {
+        const index = this._overflowItems.toArray().findIndex((item) => item === element);
+
+        if (index !== -1) {
+            this._keyboardEventsManager.setActiveItem(index);
+        }
+    }
+
+    /**
+     * Registers popover content directive for main component.
+     * @param content {OverflowPopoverContent} directive
+     */
+    registerPopoverContent(content: OverflowPopoverContent): void {
+        this._overflowPopoverContent = content;
+    }
+
+    /** @hidden */
+    _itemsTrackFn(_: number, item: OverflowItemRef): TemplateRef<any> {
+        return item.templateRef;
+    }
+
+    /** @hidden */
+    _onPopoverStateChange(opened: boolean): void {
+        if (opened) {
+            this._overflowPopoverContent?.focusFirstTabbableElement();
+        }
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._fitVisibleItems();
+        this._setFocusKeyManager();
+        this._listenToChanges();
+        this._subscribeToRtl();
+    }
+
+    /** @hidden */
+    private _listenToChanges(): void {
+        this._subscription.add(
+            this._items.changes.subscribe(() => {
+                setTimeout(() => {
```

Comment:
So. I need calculation function run when everything is being rendered. Using zone's microtaskEmpty or stable didn't worked as I expected because in some cases it simply didn't triggered. With setTimeout we ensure that recalculation will happen after current tasks are being finished.